packages/flutter/lib/src/services/raw_keyboard_macos.dart

```diff
@@ -75,14 +75,18 @@ class RawKeyEventDataMacOs extends RawKeyEventData {
     if (numPadKey != null) {
       return numPadKey;
     }
-
-    // Look to see if the keyCode is one we know about and have a mapping for.
+    // If this key is printable, generate the LogicalKeyboardKey from its unicode value.
+    // Control keys such as ESC, CRTL, and SHIFT are not printable. HOME, DEL, arrow keys, and function
+    // keys are considered modifier function keys, which generate invalid Unicode scalar values.
     if (keyLabel != null &&
-        !LogicalKeyboardKey.isControlCharacter(keyLabel)) {
+        !LogicalKeyboardKey.isControlCharacter(keyLabel) &&
+        !isFunctionKey(keyLabel)) {
```

Comment:
I still don't like `isFunctionKey` because it's too easily confused with `F1`, `F2`, etc.  Maybe `isUnprintableKey`?

Reply:
Done